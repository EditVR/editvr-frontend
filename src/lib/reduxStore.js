/**
 * @file reduxStore.js
 * Constructs Redux store and persistor.
 */

import { createStore, applyMiddleware } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import reducers from '../reducers';
import actions from '../actions';

// Set up persisted reducers using redux-persist.
const persistedReducer = persistReducer(
  {
    key: 'editvr-root',
    storage
  },
  reducers
);

// Set up Saga middleware using redux-saga.
const sagaMiddleware = createSagaMiddleware();

// Compose a store with persisted reducers, and middlewares.
const store = createStore(
  persistedReducer,
  {},
  composeWithDevTools(applyMiddleware(sagaMiddleware))
);

sagaMiddleware.run(actions);
const persistor = persistStore(store);

export { store, persistor };
