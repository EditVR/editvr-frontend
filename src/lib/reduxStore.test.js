/**
 * @file reduxStore.test.js
 * Contains tests for reduxStore.js.
 */

import { store, persistor } from './reduxStore';

describe('reduxStore()', () => {
  it('Returns correctly constructed store and persistor objects.', () => {
    expect(store).toMatchSnapshot();
    expect(persistor).toMatchSnapshot();
  });
});
