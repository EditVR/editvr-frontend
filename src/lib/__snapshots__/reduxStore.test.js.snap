// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reduxStore() Returns correctly constructed store and persistor objects. 1`] = `
Object {
  "dispatch": [Function],
  "getState": [Function],
  "replaceReducer": [Function],
  "subscribe": [Function],
  Symbol(observable): [Function],
}
`;

exports[`reduxStore() Returns correctly constructed store and persistor objects. 2`] = `
Object {
  "dispatch": [Function],
  "flush": [Function],
  "getState": [Function],
  "pause": [Function],
  "persist": [Function],
  "purge": [Function],
  "replaceReducer": [Function],
  "subscribe": [Function],
  Symbol(observable): [Function],
}
`;
